presto-main/src/main/java/com/facebook/presto/operator/BigintGroupByHash.java

```diff
@@ -278,12 +278,11 @@ private boolean tryRehash()
 
         // An estimate of how much extra memory is needed before we can go ahead and expand the hash table.
         // This includes the new capacity for values, groupIds, and valuesByGroupId as well as the size of the current page
-        preallocatedMemoryInBytes = (newCapacity - hashCapacity) * (long) (Long.BYTES + Integer.BYTES) + (calculateMaxFill(newCapacity) - maxFill) * Long.BYTES + currentPageSizeInBytes;
+        preallocatedMemoryInBytes = newCapacity * (long) (Long.BYTES + Integer.BYTES) + calculateMaxFill(newCapacity) * Long.BYTES + currentPageSizeInBytes;
```

Comment:
why `hashCapacity` and `maxFill` has been removed from this calculation?

Reply:
As mentioned in the first comment, we are now account for both of the new table (rehash target) and the old table now. `hashCapacity` and `maxFill` are denoting the number of entries in the old table, when we multiply the values with Long.BYTES and Integer.BYTES then it would become the size of the new table. As for the detail I need to check the underlying layout for the array. But overall it is why we remove `hashCapacity` and `maxFill` from deduction.